onnxruntime/test/util/test_utils.cc

```diff
@@ -96,25 +88,75 @@ void RunAndVerifyOutputsWithEP(const ORTCHAR_T* model_path, const char* log_id,
   std::vector<OrtValue> expected_fetches;
   ASSERT_STATUS_OK(session_object.Run(run_options, feeds, output_names, &expected_fetches));
 
-  auto provider_type = execution_provider->Type();  // copy string so the std::move doesn't affect us
-
   //
   // get output with EP enabled
   //
-  InferenceSessionWrapper session_object2{so, GetEnvironment()};
-  ASSERT_STATUS_OK(session_object2.RegisterExecutionProvider(std::move(execution_provider)));
-  ASSERT_STATUS_OK(session_object2.Load(model_path));
-  ASSERT_STATUS_OK(session_object2.Initialize());
 
   // make sure that some nodes are assigned to the EP, otherwise this test is pointless...
   const auto& graph2 = session_object2.GetGraph();
   auto ep_nodes = CountAssignedNodes(graph2, provider_type);
-  ASSERT_GT(ep_nodes, 0) << "No nodes were assigned to " << provider_type << " for " << model_path;
+  ASSERT_GT(ep_nodes, 0) << "No nodes were assigned to " << provider_type;
 
   // Run with EP and verify the result
   std::vector<OrtValue> fetches;
   ASSERT_STATUS_OK(session_object2.Run(run_options, feeds, output_names, &fetches));
   VerifyOutputs(output_names, expected_fetches, fetches);
+};
+
+void RunAndVerifyOutputsWithEP(const ORTCHAR_T* model_path, const char* log_id,
+                               std::unique_ptr<IExecutionProvider> execution_provider,
+                               const NameMLValMap& feeds) {
+  SessionOptions so;
```

Comment:
Can we just do a raw read the data from model_path into a string and call RunAndVerifyOutputsWithEPViaModelData instead of duplicating the implementation?

Reply:
that makes sense.. will try to do it this way.